### PR TITLE
Benchmarking Suite for Loss Functions in T5 Training

### PR DESCRIPTION
**Pull request by s1k0ra**
This suite benchmarks different loss functions in neural network training across three scenarios:
	1.	Standalone loss computation
	2.	Model forward pass
	3.	Complete training step (including gradient updates)

Supported loss functions:
	•	Standard Cross Entropy (CE)
	•	CE with Number Token Loss (NTL) using MSE
	•	CE with NTL using Wasserstein distance
	•	CE with NTL using Absolute Difference

The suite includes utility functions for generating synthetic data, timing benchmarks, and logging results. It supports experimentation with different configurations and outputs CSV reports of benchmark statistics.

### CODE REVIEW

File: benchmarking/loss_function_benchmark.py [hunk 1]
```diff
@@ -0,0 +1,669 @@
...
+        # Forward pass
+        outputs = model(
+            input_ids=input_ids,
+            attention_mask=attention_mask,
+            labels=labels
+        )
+        timer.stop("forward_pass", device)            
+
+        # Loss computation and backward pass
+        loss = loss_fn.forward(outputs.logits, labels)
```
Comment (zausin33) [inline benchmarking/loss_function_benchmark.py:365]: Loss gets already calculated inside the mode. Do you now calculate the loss twice?

**Review by zausin33 — APPROVED** [review]

**Review by jannisborn — APPROVED** [review]
LGTM 🚀
